File: src/main/java/ar/edu/unq/epers/unidad4/service/interfaces/PersonajeService.java
package ar.edu.unq.epers.unidad4.service.interfaces;

import ar.edu.unq.epers.unidad4.model.Personaje;

import java.util.Collection;

public interface PersonajeService {
    Personaje guardar(Personaje personajeNeo4J);
    Personaje recuperar(Long personajeId);
    Personaje recuperarPorNombre(String nombre);
    void recoger(Long personajeId, Long itemId);
    void amigarse(Long personajeId, Long amigoId);
    Collection<Personaje> recuperarAmigosDeMisAMigos(String nombre);
    Collection<Personaje> recuperarTodos();
    void clearAll();
}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/service/PersonajeService.kt
package ar.edu.unq.epers.unidad4.service

import ar.edu.unq.epers.unidad4.model.Personaje

interface PersonajeService {
    fun guardar(personaje: Personaje): Personaje
    fun recuperar(personajeId: Long): Personaje
    fun recuperarPorNombre(nombre: String): Personaje?
    fun recoger(personajeId: Long, itemId: Long)
    fun amigarse(personajeId: Long, amigoId: Long)
    fun amigosDeMisAmigos(nombre: String): Collection<Personaje>
    fun clearAll()
}